Quantum auto-upgrade

charlieTheBotDev opened the pull request:
Ran the auto-conversion over the files - Still need to go through and check that everything looks OK (and remove a bunch of empty `Additional Guidelines` sections) but I figure this is probably ready to merge back into the main upgrade branch now

I have kept both the manual conversion and the auto conversion - all automatic files have been called `auto.um` so we can gradually remove them moving forwards

Commit:
WIP: This is WIP based on a hack of the old quantum format

Commit:
docs(*): Remove empty files and clean up conversion

Commit:
chore(*): Remove build script and old docs

Commit:
chore(*): Remove remaining conversion scripts

Commit:
docs(*): Put manually converted docs back

Commit:
chore(*): Remove unnecessary changes to package and quantum config

Commit:
fix(transforms): Include the correct quantum module

We were including quantum-js which is the old quantum-core. It worked
because hexagon depends on quantum-js still

Commit:
docs(getting-started): Add a getting started page with a live demo

Commit:
chore(formatting): Fix some leftover quantum formatting issues

- Remove inline code languages as they are no longer supported
- Escape all codeblocks
- Fix all relative template imports
-

Commit:
docs(printing): Update the printing docs

Commit:
docs(*): De-duplicate version config and update site sidebar

Commit:
chore(auto-convert): Fix bug(s) with the auto-convert

- versions in incorrect order
- code section regex caused deletion of descriptions
- excess whitespace

Commit:
docs(plot): Fix examples where the ID matches the section id

This caused the graph to attempt to render inside the header as we had
multiple copies of the same ID on the page

Resolution was to suffix all example IDs with `-example`

Commit:
docs(guidelines): Remove all empty guideline sections

We have 4 modules with guidelines:
- Color
- Data Table
- Drawing
- Plot

Commit:
docs(*): Remove auto generated content for manually converted modules

Commit:
docs(*): Fix defaults and titles broken in auto conversion

Commit:
chore(*): fix some issues found when going through the docs and remove more excess files

charlieTheBotDev commented on content/changelog/index.um:
```diff
@@ -8,21 +8,21 @@
     @reverseVisibleList true
     @process
-      @inline ../docs/**/index.um
+      @inline ../docs/**/api.um
```
Makes sense to only include the `api` sections here (reduces the amount of stuff we have to process)

charlieTheBotDev commented on content/changelog/index.um:
```diff
@@ -8,21 +8,21 @@
-        across all the modules, converting things like @code(css)[hx-btn-positive]
-        to @code(css)[hx-positive] and converting from light/dark to
+        across all the modules, converting things like @code[hx-btn-positive]
+        to @code[hx-positive] and converting from light/dark to
```
Removed language from all inline `code` as we don't support them anymore

charlieTheBotDev commented on content/changelog/index.um:
```diff
@@ -42,19 +42,19 @@
         values has changed from this:
 
-        @codeblock js
+        @@codeblock js
```
Escaped all codeblocks to prevent issues with emails etc. being parsed as quantum

charlieTheBotDev commented on content/docs/date-picker/changelog/index.um:
```diff
@@ -0,0 +1,12 @@
+    @reverseVisibleList true
+    @inline /templates/versions.um
+    @process
+      @inline ../api.um
```
We could de-dupe this page by allowing the usage of `inline` inside `describe` blocks - with that change we could do something like this:

```
@inline /templates/changelog-page.um

@changelogContent Date Picker
   @inline ../api.um
```

charlieTheBotDev commented on content/guide/getting-started/index.um:
```diff
@@ -1,8 +1,78 @@
+          @attr href: start.html
+          @attr rel: noopener
+          @attr target: _blank
+          View demo in new window
```
Added the Getting Started page

charlieTheBotDev commented on content/guide/getting-started/start.html:
```diff
@@ -0,0 +1,105 @@
+        .add(statusCard('Instance 3', 'negative')))
+  </script>
+</body>
+</html>
```
This file is used in a codeblock on the getting started page as well as in an iframe (with a link provided to open in a new page)

charlieTheBotDev commented on content/guide/installation/index.um:
```diff
@@ -45,9 +46,75 @@
     Save this in a file, then run it with node to generate your build. This api
     can be easily integrated with gulp or grunt, or can even be run as an npm task.
+
+    @api Builder API
```
Added the API for `hexagon.[theme].build`

charlieTheBotDev commented on content/guide/printing/index.um:
```diff
@@ -1,44 +1,70 @@
-@inline ../../templates/sidebar-page.um
+@inline /templates/sidebar-page.um
```
Updated the printing page to use new quantum format instead of manual `p` and `div` elements

charlieTheBotDev commented on content/modules/tag-input/versions/1.13.0.um:
```diff
@@ -1,11 +0,0 @@
-@version 1.13.0
```
Deleted `content/modules` (the old docs)

charlieTheBotDev commented on content/templates/sidebar-page.um:
```diff
@@ -27,32 +27,104 @@
             @page /guide/printing: Print Styles
 
+          @section Css Components
+            @page /docs/base/: Base
```
Added the modules to the sidebar page

charlieTheBotDev commented on content/versions.json:
```diff
@@ -1,102 +0,0 @@
-{
-  "targetVersions": [
```
Removed need for this file - we get `latest` from `package.json` (hexagon version) and we use the `templates/versions.um` file instead of passing in config

charlieTheBotDev commented on old_module_docs/date-picker/index.um:
```diff
@@ -1,2 +0,0 @@
-@module Date Picker
-  @inline versions/*.um
```
Deleted `old_module_docs`

charlieTheBotDev commented on quantum.config.js:
```diff
@@ -115,6 +116,11 @@ module.exports = {
+    {
+      files: 'content/guide/**/*.html',
+      base: 'content',
+      watch: false
```
Include the `guide` html so we can have iframe examples easily (without having to go through quantum to generate the HTML)

charlieTheBotDev commented on transforms/transforms.js:
```diff
@@ -1,4 +1,4 @@
-const quantum = require('quantum-js')
+const quantum = require('quantum-core')
```
Fixes the issue where example codeblocks didn't render - it was due to using `quantum.select` from the old quantum which doesn't satisfy `looksLikeAnEntity`

jmsmyth replied:
How did this require even work...

charlieTheBotDev replied:
Hexagon-js depends on quantum-js so the module was there.

The old selection api was similar enough not to break the new transformer :P

jmsmyth commented:
Looks good to me

Commit:
docs(*): De-version descriptions

We only have one module with versions so it didn't make sense to version
them all

Also fixed some find/replace bugs with codeblocks
